crates/language/src/language.rs

```diff
@@ -310,6 +310,14 @@ pub trait LspAdapterDelegate: Send + Sync {
     async fn shell_env(&self) -> HashMap<String, String>;
     async fn read_text_file(&self, path: &RelPath) -> Result<String>;
     async fn try_exec(&self, binary: LanguageServerBinary) -> Result<()>;
+
+    fn worktree_root_dir(&self) -> PathBuf {
+        let mut path = self.worktree_root_path().to_path_buf();
+        if path.is_file() {
```

Comment:
This should go through our fs abstraction instead as `path.is_file` is a blocking io call. Instead of adding this method to the trait I think we can just do something like
```rust
let mut worktree_abs_path = self.worktree.abs_path();
if self.fs.is_file(&worktree_abs_path).await {
...
```
in the two call sites below